server/src/auth/token_validator.rs

```diff
@@ -91,12 +92,54 @@ impl TokenValidator {
             Ok(updated_tokens)
         }
     }
+
+    pub async fn schedule_validation_of_known_tokens(&self, validation_interval_seconds: u64) {
+        let sleep_duration = tokio::time::Duration::from_secs(validation_interval_seconds);
+        loop {
+            tokio::select! {
+                _ = tokio::time::sleep(sleep_duration) => {
+                    let _ = self.revalidate_known_tokens().await;
+                }
+            }
+        }
+    }
+    pub async fn revalidate_known_tokens(&self) -> EdgeResult<()> {
+        let tokens_to_validate: Vec<String> = self
+            .token_cache
+            .iter()
+            .filter(|t| t.value().status == TokenValidationStatus::Validated)
```

Comment:
Crazy idea: maybe we can validate all known tokens. Perhaps some were forbidden to access something and then re-added (because of reasons). We might revive the token on the next request that tries to use it... but I'm not so sure

Reply:
That feels risky with how many tokens that might end up revalidating.

Since we're now persisting only validated tokens, restarting edge will give you another attempt with an earlier forbidden token.

Since the default token generation process generates random strings/secrets, the chances of you deploying an app with a not-generated token and then going to generate it are low.